### ISSUE TITLE
Error in trainer when schedule is OneCyclelr.  "Tried to step {} times. The specified number of total steps is {}"

### ISSUE THREAD

**Issue by fernando080** [flairNLP/flair#2954]
### The Bug
In the common use of OneCyclelr during a training process it won't present any error, but when you set **train_with_test** (as you do with dev), is necesary to recalculate the total number of steps due to the dataset_size have been increased. You can check it in the screenshot below.

### Code to Reproduce
I think you coul reproduce it with a simple training set and the Sequencetagger. 
`
tagger: SequenceTagger = SequenceTagger(
                    hidden_size=...,
                    embeddings=...,
                    tag_dictionary=...,
                    tag_type='ner',
                    use_crf=...,
                    use_rnn=...,
                    reproject_embeddings=...,
                )

trainer = ModelTrainer(tagger, corpus, optimizer=...)

from torch.optim.lr_scheduler import OneCycleLR

trainer.train(f"resources/ner",
                    learning_rate=...,
                    mini_batch_size=...,
                    mini_batch_chunk_size=...,
                    max_epochs=...,
                    scheduler=OneCycleLR,
                    weight_decay=...,
                    train_with_test=True,
                )
`

### Screenshots
![image](https://user-images.githubusercontent.com/82539263/193404823-dc3f60f0-d880-4394-a991-883c072c78a8.png)

**Environment **
 - Windows 10
 - Version flair 0.10.0:

It is really easy to fix, I can solve this issue if it's okay with you
